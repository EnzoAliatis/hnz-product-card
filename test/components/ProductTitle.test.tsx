import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product } from '../data/products';

describe('Product Title', () => {

  test('should show the component well mounted', () => {
    const wrapper = renderer.create(
      <ProductTitle title='custom product' className='custom-class' />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show the component with the name of the product', () => {

    const wrapper = renderer.create(
      <ProductCard product={product}>
        {
          () => (
            <ProductTitle />
          )
        }

      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
})