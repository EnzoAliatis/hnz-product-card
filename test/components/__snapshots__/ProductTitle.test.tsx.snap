// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Title should show the component well mounted 1`] = `
<span
  className="productDescription custom-class"
>
  custom product
</span>
`;

exports[`Product Title should show the component with the name of the product 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - no image
  </span>
</div>
`;
